report-jtreg-plugin/src/main/java/io/jenkins/plugins/report/jtreg/BuildReportExtendedPlugin.java

```diff
@@ -25,14 +25,86 @@
         super(buildNumber, buildName, passed, failed, error, suites, addedSuites, removedSuites, testChanges, total, notRun, allTests, job);
+        this.job = job;
+    }
+
+    public List<ComparatorLinksGroup> getMatchedComparatorLinksGroups() {
+        List<ComparatorLinksGroup> matchedComparatorLinksGroup = new ArrayList<>();
+        for (ComparatorLinksGroup link : JenkinsReportJckGlobalConfig.getGlobalComparatorLinksGroups()) {
+            if (job.matches(link.getJobMatchRegex())) {
+                matchedComparatorLinksGroup.add(link);
+            }
+        }
+
+        return matchedComparatorLinksGroup;
+    }
+
+    public String createComparatorLinkUrl(LinkToComparator ltc) {
+        StringBuilder url = new StringBuilder(getCompUrlStub());
+
+        for (String arg : ltc.getComparatorArguments().split("\n")) {
+            if (arg.matches("^-+regex.*")) {
+                url.append("--regex+");
+                url.append(parseToRegex(ltc.getSpliterator(), arg.split(" ")[1])); // split the arg my space and give it the second part
+            } else {
+                url.append(arg);
+                url.append("+");
+            }
+        }
+
+        return url.toString()
+                .replace(" ", "+")
+                .replace("#", "%23")
+                .replace(".", "%2E");
+    }
+
+    private String parseToRegex(String spliterator, String query) {
+        String[] splitJob = job.split(spliterator);
+
+        String converted = query;
+
+        // finds %N in the query from Jenkins config and replaces it with corresponding part of job name
+        Pattern p = Pattern.compile("%\\{((N-)?[0-9]+|S|SPLIT)\\}");
+        Matcher m = p.matcher(converted);
 
+        while (m.find()) {
+            String insideBrackets = converted.substring(m.start() + 1 + 1, m.end() - 1); // get just the inside of the brackets
+
+            String replacement;
+            if (insideBrackets.equals("S") || insideBrackets.equals("SPLIT")) {
+                replacement = spliterator;
+            } else {
+                int number;
+                if (insideBrackets.charAt(0) == 'N') {
+                    number = splitJob.length + Integer.parseInt(insideBrackets.substring(1));
+                } else {
+                    number = Integer.parseInt(insideBrackets) - 1;
+                }
+
+                if (number < 0 || number >= splitJob.length) {
+                    System.err.println("WARNING: The number given in the --regex argument of \"Comparator links\" section is out of range of the job name!");
+                    return "The number given is out of range of the job name!";
+                }
+
+                replacement = splitJob[number];
+            }
+
+            converted = converted.replaceFirst("%\\{((N-)?[0-9]+|S|SPLIT)\\}", replacement);
```

Comment:
Dont forget the replace is eager. I bet we will find (in unittest O:) ) something where the eagerness maybe bite us.

Otherwise  the algortihm looks good. why is there the converted=converted.repalceFirst? I thought that the matcher will aloow you to do this. 
if this line will persits, then you have to remove duplicated regex declaration.